pkg/systemd/services/timer-dialog.jsx

```diff
@@ -233,27 +248,58 @@ const CreateTimerDialogBody = ({ owner }) => {
                                         }}
                                         aria-label={_("Repeat")}>
                                 <FormSelectOption value="no" label={_("Don't repeat")} />
+                                <FormSelectOption value="minutely" label={_("Minutely")} />
                                 <FormSelectOption value="hourly" label={_("Hourly")} />
                                 <FormSelectOption value="daily" label={_("Daily")} />
                                 <FormSelectOption value="weekly" label={_("Weekly")} />
                                 <FormSelectOption value="monthly" label={_("Monthly")} />
                                 <FormSelectOption value="yearly" label={_("Yearly")} />
+                                <FormSelectOption value="custom" label={_("Custom")} />
                             </FormSelect>
                         </FormGroup>
                         {repeat == "no" &&
                         <FormGroup label={_("Run at")}>
                             <TimePicker className="create-timer-time-picker specific-no-repeat"
                                         menuAppendTo={() => document.body} time={specificTime} is24Hour onChange={setSpecificTime} />
                         </FormGroup>}
+                        {repeat == "custom" &&
+                        <FormGroup
+                            label={_("Custom")}
+                            validated={submitted && validateCustom ? "error" : "default"}
+                            helperTextInvalid={_("Invalid OnCalendar Event")}>
+                            <Flex>
+                                <TextInput
+                                    id="custom"
+                                    className='custom-group'
+                                    value={custom}
+                                    onChange={setCustom}
+                                    validated={submitted && validateCustom ? "error" : "default"}
+                                />
+                                <Popover key="popover-unix-group-custom" id="popover-unix-group-custom" position={ PopoverPosition.right }
+                                        headerContent={<div>Custom Timer Details</div>}
```

Comment:
`Custom timer details`, no sentance case. Also it needs to be localized, so writing it as `headerContent={_("Custom timer details")}.
Also not sure if that is the best title or if the title is actually required.
![Screenshot from 2022-07-27 10-23-16](https://user-images.githubusercontent.com/12330670/181198889-3ce1965c-13e9-486a-ae7d-5ce242c0c355.png)

@garrett WDYT of the text?

Reply:
- I missed the custom options. Whoops. (Blame the morning, coffee not kicking in yet, etc.) This isn't acceptable (yet).
- I don't understand what this is supposed to be saying. Is it supposed to be something like this? "Use systemd onCalendar formatting."
- Why does it say "IE"? Is that part of the string? If you're trying to say "i.e." (lowercase, periods needed); that's wrong as it means "that is", it's supposed to be "e.g.", which means "for example"… but please write out "Example:" instead.
- However, we normally DO NOT use popovers for this. It would be better as helper text here, so people can actually refer to it while typing and do not need to click on something extra to see what to do.

Mockup:

![create-timer-custom excalidraw(2)](https://user-images.githubusercontent.com/10246/181209399-bd153294-5f86-461d-b4cf-9904c9e02a73.png)

Reply:
Would also link fit into the description?

Reply:
Helper text guidelines: https://www.patternfly.org/v4/components/helper-text/design-guidelines/#when-to-use-helper-text

When to use helper text:
> Use helper text to provide imperative information regarding an input, such as character limits, password expectations, and more.

When to use a popover instead of helper text:
> Do not use helper text to explain what a form field is or to provide additional form context. Instead, use a popover to convey any additional, non-imperative information to a user.

When to use placeholder text instead of helper text:
> Do not use helper text to provide formatting examples within a text box. Instead, use placeholder text to temporarily showcase what content should appear in a field before the user inputs information. 

This kind of information falls under the first point, to use helper text, as it's important, necessary information to understand what "custom" means and has an example that needs to be referred to while typing. Also, the placeholder text would go away, so having an example for a specific format versus something simple (like a name) needs to stay put to be referenced.

And, as I mentioned above, showing that information overlapping with where you need it is a problem... additionally, it's an accessibility concern as it changes the context, which makes it tricky for screen readers and extremely difficult for multiple cognitive disabilities (such as anyone with attention issues, "brain fog", short term memory problems, and so on).

I'm just trying to clarify why it's better to use this style instead of the popover. 

Additionally, we also have the space here, so let's use it.

Reply:
> Would also link fit into the description?

https://www.patternfly.org/v4/components/helper-text/design-guidelines/#when-to-use-helper-text

> Include hyperlinks to additional information if necessary

If it's offsite (which I assume a link would be), then we need an offsite icon.

We do use offsite links to more information in other places in Cockpit, including helperText areas on forms like this.

So: Yes.

Reply:
switched to helper text as suggested.